build/source/engine/t2enthalpy.f90

```diff
@@ -0,0 +1,468 @@
+  if(err/=0)then; err=20; message=trim(message)//'problem allocating lookup table var dimension dimension'; return; end if
+
+  ! allocate space for the values in the lookup table
+  do iVar=1,maxvarLookup
+   if(allocated(lookup_data%z(iSoil)%var(iVar)%lookup))then; check=.true.; else; allocate(lookup_data%z(iSoil)%var(iVar)%lookup(nLook), stat=err); endif
+   if(check) then; err=20; message=trim(message)//'lookup table value dimension was unexpectedly allocated already'; return; end if
+   if(err/=0)then; err=20; message=trim(message)//'problem allocating lookup table vaule dimension dimension'; return; end if
+
+  end do ! (looping through variables)
+ end do ! (looping through soil layers)
+
+ ! loop through soil layers
+ do iSoil=1,nSoil
+
+  ! -----
+  ! * make association to variables in the data structures...
+  ! ---------------------------------------------------------
+
+  associate(&
+
+  ! associate model parameters
+  snowfrz_scale  => mpar_data%var(iLookPARAM%snowfrz_scale)%dat(1)           , & ! scaling parameter for freezing     (K-1)
+  soil_dens_intr => mpar_data%var(iLookPARAM%soil_dens_intr)%dat(iSoil)      , & ! intrinsic soil density             (kg m-3)
+  theta_sat      => mpar_data%var(iLookPARAM%theta_sat)%dat(iSoil)           , & ! soil porosity                      (-)
+  theta_res      => mpar_data%var(iLookPARAM%theta_res)%dat(iSoil)           , & ! volumetric residual water content  (-)
+  vGn_alpha      => mpar_data%var(iLookPARAM%vGn_alpha)%dat(iSoil)           , & ! van Genuchten "alpha" parameter    (m-1)
+  vGn_n          => mpar_data%var(iLookPARAM%vGn_n)%dat(iSoil)               , & ! van Genuchten "n" parameter        (-)
+
+  ! associate values in the lookup table
+  Tk            => lookup_data%z(iSoil)%var(iLookLOOKUP%temperature)%lookup  , & ! temperature (K)
+  Ey            => lookup_data%z(iSoil)%var(iLookLOOKUP%enthalpy)%lookup     , & ! enthalpy (J m-3)
+  E2            => lookup_data%z(iSoil)%var(iLookLOOKUP%deriv2)%lookup         & ! second derivative of the interpolating function
+
+  ) ! end associate statement
+
+  ! compute vGn_m
+  vGn_m = 1._dp - 1._dp/vGn_n
+
+  ! -----
+  ! * populate the lookup table...
+  ! ------------------------------
+
+  ! initialize temperature and enthalpy
+  Tk(nLook) = Tfreeze
+  Ey(nLook) = 0._dp
+
+  ! loop through lookup table
+  do iLook=(nLook-1),1,-1
+
+   ! update temperature and enthalpy
+   Tk(iLook) = Tk(iLook+1)
+   Ey(iLook) = Ey(iLook+1)
+
+   ! get the temperature increment for the numerical integration
+   T_incr = (xTemp(iLook)-xTemp(iLook+1))/real(nIntegr8, kind(dp))
+
+   ! numerical integration between different values of the lookup table
+   do jIntegr8=1,nIntegr8
+
+    ! update temperature
+    Tk(iLook)  = Tk(iLook) + T_incr
+
+    ! compute the volumetric liquid water and ice content
+    ! NOTE: assume saturation
```

Comment:
Is this a safe assumption? Are there reasons why we don't compute this on each time step with the current values for vFracLiq and vFraqIce?